### ISSUE TITLE
Table sort function doesn't show chevron

### ISSUE THREAD

**Issue by pieqq** [canonical-web-and-design/vanilla-framework#2661]
Hello!

**Describe the bug**
The Vanilla documentation about Tables mentions an option to [sort the rows.](https://docs.vanillaframework.io/base/tables/#sortable) However, the example doesn't work for me (on neither Firefox 70.0.1 nor Chromium 78.0.3904.97). I get the same result when trying to implement a sample HTML file locally.

This issue is probably related to #2634 and #2635 .

**To Reproduce**
1. Go to https://docs.vanillaframework.io/base/tables/#sortable
2. Click on one of this table headers (for instance, "RAM").
3. On 1st click, nothing happens. On 2nd click, the rows are sorted descending. On 3rd click, the rows are sorted ascending. On 4th click, nothing happens, etc.
4. Moreover, there are no chevron icon to show in what sort mode it currently is.

**Expected behavior**
As stated by the page, when clicking on a header,

> it will change the chevron icon in that direction.

Because of this, the behavior appears weird (in the example given, the 1st click seems to do nothing when it actually sets the `aria-sort` attribute to `ascending`). Maybe using a table with more rows and more different information amongst rows would be beneficial.

I also noticed that the introduction paragraph says

> Using the class p-table--sortable and assigning role="columnheader" and aria-sort to each <th> ...

but the actual example source code doesn't make use of the `role="columnheader"` part:

```html
      <th id="t-name" aria-sort="none">Status</th>
      <th id="t-users" aria-sort="none" class="u-align--right">Cores</th>
      <th id="t-units" aria-sort="none" class="u-align--right">RAM</th>
      <th id="t-revenue" aria-sort="none" class="u-align--right">Disks</th>
```

**Screenshots**
![vanilla-sortable-table](https://user-images.githubusercontent.com/512471/69313833-9c366300-0c6d-11ea-9c7e-e4d0f7f9135f.gif)

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux 18.04
 - Browser Firefox 70, Chromium 78
 - Version 2.4.1

Thanks! :)

**Comment by pieqq** [comment]
OK I found an answer to my issue: it is indeed because of the missing `role="columnheader"` that the chevron is not shown. If I modify the HTML of the documentation page using Firefox dev tools to add `role="columnheader"` too all `<th>`, I get a visual indication that the header is clickable and the chevron.

**Comment by anthonydillon** [comment]
@pieqq thank you for taking the time to report this issue. You are correct the styling of the sorting is applied via the role which was mistakenly removed and has been added back in an [earlier PR](https://github.com/canonical-web-and-design/vanilla-framework/pull/2640). 

The functionality is due to a difference it the `sort` function between browsers. The current build works on Firefox but not Chrome. I have addressed this in [this PR](https://github.com/canonical-web-and-design/vanilla-framework/pull/2667). 

These fixes will be available in the next release due soon.
